src/key.h

```diff
@@ -183,7 +183,10 @@ struct CExtKey {
     template <typename Stream>
     void Unserialize(Stream& s)
     {
-        unsigned int len = ::ReadCompactSize(s);
+        unsigned int len = ::ReadCompactSize(s); //not using BIP32_EXTKEY_SIZE as max size here -> want to throw my own exception
```

Comment:
ReadCompactSize returns uint64_t, for 64-bit it's the same as size_t.... and s.read expects size_t parameter, so:
const size_t len = ::ReadCompactSize(s);